### DISCUSSION TITLE
我想要打包它

### DISCUSSION THREAD

**Discussion by King-XVX** [community/community#143647] [Codespaces]
### Select Topic Area

Question

### 打包它

import pygame
import turtle as t

pygame.init()
screen = pygame.display.set_mode((750, 750))

zuobianyuan = 25
youbianyuan = 725
dingbubianyuan = 25
dibubianyuan = 25
kuan = 50
gaodu = 50

m = [0]*16
for i in range(16):
    m[i] = [0]*16

player = 1
winner = 0

running = True

def check(row, col):

    # 左右方向判断
    score = 1
    for i in range(4):
        if m[row][col + i] == m[row][col + i + 1]:
            score += 1
        else:
            break
    for i in range(4):
        if m[row][col - i] == m[row][col - i - 1]:
            score += 1
        else:
            break
    if score >= 5:
        return True

    # 上下方向判断
    score = 1
    for a in range(4):
        if m[row + a][col] == m[row + a + 1][col]:
            score = score+1
        else:
            break
    for a in range(4):
        if m[row - a][col] == m[row - a - 1][col]:
            score = score+1
        else:
            break
    if score >= 5:
        return True

    # 左下到右上判断
    score = 1
    for i in range(4):
        if m[row + i][col+i] == m[row + i + 1][col + i + 1]:
            score += 1
        else:
            break
    for i in range(4):
        if m[row - i][col - i] == m[row - i - 1][col - i - 1]:
            score += 1
        else:
            break

    if score >= 5:
        return True

    # 左上到右下判断
    score = 1
    for i in range(4):
        if m[row - i][col + i] == m[row - i - 1][col + i + 1]:
            score += 1
        else:
            break
    for i in range(4):
        if m[row + i][col - i] == m[row + i + 1][col - i - 1]:
            score += 1
        else:
            break

    if score >= 5:
        return True

while running:
    for event in pygame.event.get():
        if event.type == pygame.QUIT:
            running = False
        elif event.type == pygame.MOUSEBUTTONDOWN:
            x, y = pygame.mouse.get_pos()
            col = round((x - 25)/50)
            row = round((y - 25)/50)
            if m[row][col] == 0:
                print(row, col)

                m[row][col] = player
                if (check(row, col)):
                    winner = player
                    if (winner != 0):
                        if winner == 1:
                            winner = '黑子赢了'

                        elif winner == 2:
                            winner = '白子赢了'

                        print(f'{winner}！！！')

                else:
                    if player == 1:
                        player = 2
                    else:
                        player = 1
            else:
                print('这有棋子了！')

    screen.fill('#EE9A49')
# 竖线
    for x in range(15):
        pygame.draw.line(screen, '#000000', [zuobianyuan+kuan*x, dingbubianyuan], [zuobianyuan+kuan*x, youbianyuan], 2)
# 横线
    for y in range(15):
        pygame.draw.line(screen, '#000000', [zuobianyuan, dingbubianyuan + gaodu*y], [youbianyuan, dingbubianyuan + gaodu*y], 2)

    pygame.draw.circle(screen, '#000000', [25+50*7, 25+50*7], 8)

    x,y = pygame.mouse.get_pos()
    x = round((x - zuobianyuan) / kuan) * kuan + zuobianyuan
    y = round((y - dingbubianyuan) / gaodu) * gaodu + dingbubianyuan

    pygame.draw.rect(screen, '#FFFFFF', [x-25, y-25, 50, 50], 2)

    # pygame.draw.rect(screen, (153, 51, 250), [50, 750, 100, 50], 2) 没什么用，本来是用来作模式选择的按钮角色绘制，但是没有字的包，所以分了两个代码组

    for row in range(15):
        for col in range(15):
            if m[row][col] == 1:
                pygame.draw.circle(screen, '#000000', [col*kuan+zuobianyuan, row*gaodu+dingbubianyuan], 25)
            elif m[row][col] == 2:
                pygame.draw.circle(screen, '#FFFFFF', [col * kuan + zuobianyuan, row * gaodu + dingbubianyuan], 25)

    pygame.display.update()

    if (winner!=0):
        if winner == 1:
            text_turtle = t.Turtle()
            text_turtle.hideturtle()
            text_turtle.write(f'{winner},三秒后关闭', align='center', font=('Arial', 60, 'bold'))
        else:
            text_turtle = t.Turtle()
            text_turtle.hideturtle()
            text_turtle.write(f'{winner},三秒后关闭', align='center', font=('Arial', 60, 'bold'))
        pygame.display.update()
        pygame.time.wait(3000)
        running = False

pygame.quit()

这是一段我闲暇时间制作的游戏，我想要打包它发给我的同学，但是我用的auto-py-to-exe无法打包它，谁可以帮助我。哦！它要是不去翻译那些变量该多好呀。

This is a game I made in my spare time, I want to package it and send it to my classmates, but the auto-py-to-exe I used can't package it, who can help me? Ouch! If only it didn't have to translate those variables.

**Comment by mslxl** [comment]
Well, maybe you should learn markdown first. Your code format was messy.

也许，你应该先学习一下 markdown？你的代码的格式掺成了面条。

For your question: I do not know about packaging python script, but you can try to use pyinstaller, my senior schoolfellow has developed a python application and he use pyinstaller, it could work without python env installed.

对于你的问题：我对python应用的打包了解并不多，不过我推荐使用 pyinstaller。我的学长曾经使用 pyinstaller 打包过一个应用程序，它可以在没有预安装 python 的环境下运行。

PS: Seems you are Chinese, but your language is pretty strange. It just like translationese. I'd recommend you to use English title in English communities, not everyone could know kanji in these communities. Besides, do not send duplicated content.

PS: 看你的 profile 似乎是中国人？但是你的中文看上去有点怪，有那么亿点点像翻译腔。我推荐在英文社区中使用英文标题，因为这个社区里并不是所有人都能看懂汉字。此外，不要发送重复内容。

**Comment by Akash1134** [comment]
Hi @King-XVX , thanks for participating!

We take our [Code of Conduct](https://github.com/community/community/blob/0ca37cf594e438fd4aec3c7f3736b95e6b0b3567/CODE_OF_CONDUCT.md) very seriously and want to help ensure that everyone has a good experience free of antagonism and harassment. Unfortunately, we don’t currently have moderators for languages other than English. Until that changes, we need to ask that everyone use English here in the GitHub Community when posting. We’ll be locking any posts in languages other than English for now, including this one.
